app/rust/src/parser/utils.rs

```diff
@@ -30,3 +30,51 @@ pub fn read_varint(input: &[u8]) -> Result<(&[u8], u64), nom::Err<ParserError>>
         _ => Ok((rem, prefix as _)),
     }
 }
+
+pub trait ApduPanic: Sized {
+    type Item;
+
+    fn apdu_unwrap(self) -> Self::Item;
+
+    fn apdu_expect(self, s: &str) -> Self::Item;
+}
+
+impl<T, E> ApduPanic for Result<T, E> {
+    type Item = T;
+
+    #[inline]
+    fn apdu_unwrap(self) -> Self::Item {
+        match self {
+            Ok(t) => t,
+            Err(_) => panic!(),
```

Comment:
actually, if you are still tight, it would be _better_ if you replace these with `core::hint::unreachable_unchecked`, that hints the compiler to remove the branch entirely, saving a lot of space, not only due to the removed branch, but the removed panic machinery!

Reply:
https://github.com/Zondax/ledger-rust/blob/feat/v2/bolos/src/panic_traits.rs